services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -294,12 +310,21 @@ public Builder filterExpression(Expression filterExpression) {
          * "https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html"
          * >Accessing Item Attributes</a> in the <i>Amazon DynamoDB Developer Guide</i>.
          * </p>
-         * @param attributesToProject
-         *        A collection of the attributes names to be retrieved from the database.
+         *
+         * @param attributesToProject A collection of the attributes names to be retrieved from the database.
          * @return Returns a reference to this object so that method calls can be chained together.
          */
         public Builder attributesToProject(Collection<String> attributesToProject) {
-            this.attributesToProject = attributesToProject != null ? new ArrayList<>(attributesToProject) : null;
+            //If there is a reset then clear the attributesToProject List.
+            if (attributesToProjectStringList != null) {
+                attributesToProjectStringList
+                        .forEach(attribute -> this.attributesToProject.remove(NestedAttributeName.create(attribute)));
+            }
```

Comment:
I think this is really confusing. We don't want to treat these as separate lists. We should make it clear in the javadoc that addNestedAttributeToProject is adding a 'nested attribute' to the 'attributesToProject' list. Let's make it clear to the customer there is just one list, and let's treat it like one list.

Reply:
I will rename attributesToProjectStringList to ** attributesToProjectTempList** , the intention of this list is to keep the backward compatibility , earlier if the user called attributesToProject() twice it used to reset the old values with new one we need to keep this behavior.